frontend/src/features/public-form/components/FormPaymentPage/FormPaymentResumeModal.tsx

```diff
@@ -0,0 +1,93 @@
+import { useNavigate } from 'react-router-dom'
+import {
+  ButtonGroup,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+  useDisclosure,
+} from '@chakra-ui/react'
+
+import { FormResponseMode } from '~shared/types'
+
+import { useBrowserStm } from '~hooks/payments'
+import { useIsMobile } from '~hooks/useIsMobile'
+import { getPaymentPageUrl } from '~utils/urls'
+import Button from '~components/Button'
+
+import { usePublicFormContext } from '../../PublicFormContext'
+
+/**
+ * This component is split up so that input changes will not rerender the
+ * entire FormFields component leading to terrible performance.
+ */
+export const PublicFormPaymentResumeModal = (): JSX.Element => {
+  const isMobile = useIsMobile()
+  const { form, formId } = usePublicFormContext()
+
+  const [lastPaymentMemory, , clearPaymentMemory] = useBrowserStm(formId)
+
+  const isPaymentEnabled =
+    form?.responseMode === FormResponseMode.Encrypt &&
+    form?.payments_field?.enabled
+
+  const { isOpen, onClose } = useDisclosure({
+    defaultIsOpen: Boolean(lastPaymentMemory && isPaymentEnabled),
+  })
+
+  const navigate = useNavigate()
+  if (!isOpen) {
+    return <></>
+  }
+  const onSubmit = () => {
+    if (!lastPaymentMemory) {
+      onClose()
+      return
+    }
+    navigate(getPaymentPageUrl(formId, lastPaymentMemory.paymentId))
+  }
+
+  const handleStartOver = () => {
+    clearPaymentMemory()
+    onClose()
+  }
+  return (
+    <Stack px={{ base: '1rem', md: 0 }} pt="2.5rem" pb="4rem">
+      <Modal
+        isOpen
+        onClose={() => {
+          // do nothing, prevent dismissal through backdrop touch
+        }}
+      >
+        <ModalOverlay />
```

Comment:
Slight UI margin inconsistencies with Design:

modal header to body: 16px  -> 32px(2rem)
modal body to footer: 32px -> 40px(2.5rem)
modal footer to end of modal: 44px -> 40px(2.5rem)

Reply:
Updated here c3a260d3

Side note: I thought our semantic components would be consistent with the design 🤔

Reply:
Yeah cause we directly using the modal components from chakra UI

Not urgent but since recently we have been using a lot of Modals think we should probably wrap modal components like what we did for other components in `frontend/src/components`, currently we only have a `ModalCloseButton` 

Should probably make it consistent with our figma!